### PR TITLE
Add missing JDBC driver configuration for MS SQL Server

### LINKED ISSUE #15060: Transaction deadlock with Microsoft SQL if "sendStringParametersAsUnicode=false" not set in db url properties
### Describe the bug

Transaction deadlock error occurs when concurrently creating more than two new users with Keycloak API. Tried with Keycloak versions 18.0.1 and 19.0.3 while using Microsoft SQL Server database.

After setting `sendStringParametersAsUnicode=false` in `KC_DB_URL_PROPERTIES` there were no issues. [Database configuration instructions](https://www.keycloak.org/server/db) for Microsoft SQL Server database however states: 
> No special settings of JDBC driver or database is necessary.

There was issue on Jira backlog for updating Microsoft SQL Server database driver instructions and request in the comments to recreate bug on GitHub.
https://issues.redhat.com/browse/KEYCLOAK-19414

### Version

18.0.1, 19.0.3

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by sguilhen**
Closes #15060

### CODE REVIEW

**Review by vramik — DISMISSED** [review]
Thanks @sguilhen

File: docs/guides/src/main/server/db.adoc [hunk 1]
```diff
@@ -90,7 +90,9 @@ For details on these properties and any performance implications, see the Oracle
 
 === Unicode support for a Microsoft SQL Server database
 
-Unicode characters are supported only for the special fields for a Microsoft SQL Server database. The JDBC driver and database require no special settings.
+Unicode characters are supported only for the special fields for a Microsoft SQL Server database. The database requires no special settings,
+but the `sendStringParametersAsUnicode` property needs to be set to `false` in the JDBC driver. This is needed to prevent deadlocks that can
+happen during data type conversion.
```
Comment (hmlnarik) [inline docs/guides/src/main/server/db.adoc:95]:
```suggestion
Unicode characters are supported only for the special fields for a Microsoft SQL Server database. The database requires no special settings.

The `sendStringParametersAsUnicode` property of JDBC driver should be set to `false` to significantly
improve performance. Without this parameter, the Microsoft SQL Server might be unable to use indexes.
```

Comment (sguilhen) [reply docs/guides/src/main/server/db.adoc]: That does look a lot better. Thanks @hmlnarik , updated the PR as per your suggestion.

**Review by hmlnarik — APPROVED** [review]
Thank you for the PR!

**Review by vramik — APPROVED** [review]

**Pull request merged**